Tests/SwiftAlgorithmsTests/TestUtilities.swift

```diff
@@ -62,7 +63,30 @@ func XCTAssertEqualSequences<S1: Sequence, S2: Sequence>(
   _ message: @autoclosure () -> String = "",
   file: StaticString = #file, line: UInt = #line
 ) rethrows where S1.Element == S2.Element {
-  try XCTAssert(expression1().elementsEqual(expression2(), by: areEquivalent), message(), file: file, line: line)
+
+  func fail(_ reason: String) {
+    XCTFail(message().isEmpty ? reason : "\(message()) - \(reason)",
+            file: file, line: line)
+  }
+
+  var iter1 = try expression1().makeIterator()
+  var iter2 = try expression2().makeIterator()
+  var idx = 0
+  while true {
+    switch (iter1.next(), iter2.next()) {
+    case let (e1?, e2?) where areEquivalent(e1, e2):
+      idx += 1
+      continue
+    case let (e1?, e2?):
+      fail("unequal elements at index \(idx) (\(e1) != \(e2))")
```

Comment:
Just a wording suggestion 
```suggestion
      fail("element \(e1) on first sequence does not match element \(e2) on second sequence at position \(idx)")
```
Is is a bit more verbose but `unequal` here may not be the word since `areEquivalent` is not necessarily equality :)
Also, index could be confusing because a sequence doesn't actually have the concept of indexing ... but also not sure `position` is better

Reply:
Awesome! Thanks Luciano.

Reply:
Agreed on the "position" wording the avoid confusion with existing index terminology 🙂